src/diff_file.c

```diff
@@ -328,16 +328,25 @@ static int diff_file_content_load_workdir_file(
 	git_filter_list *fl = NULL;
 	git_file fd = git_futils_open_ro(git_str_cstr(path));
 	git_str raw = GIT_STR_INIT;
+	git_object_size_t new_file_size = 0;
 
 	if (fd < 0)
 		return fd;
 
-	if (!fc->file->size)
-	    error = git_futils_filesize(&fc->file->size, fd);
+	error = git_futils_filesize(&new_file_size, fd);
 
-	if (error < 0 || !fc->file->size)
+	if (error < 0 || !new_file_size)
```

Comment:
I don't think that I understand this test...  it seems like we could have truncated the file to zero bytes and do not treat that as an error.  Should we not always treat a non-zero `fc->file->size` and a differing `fc->file->size` to `new_file_size` as an error?  I do think it makes sense to skip `mmap`ing the zero byte case, but I think that we need to do this below the test beneath it...

```suggestion
	if (error < 0 )
```

Reply:
I agree we should skip this test. I don't think the original test made sense here either (testing fc->file->size if git_futils_filesize is successful).
Plus, removing this test ensures that we correctly handle a file becoming empty. (I'll add another test case for this soon)